src/contract/event.h

```diff
@@ -253,34 +260,10 @@ class EventManager {
     bool matchTopics(const Event& event, const std::vector<Hash>& topics = {}) const;
 
     /**
-     * Register the event in the temporary list.
-     * Keep in mind the original Event object is MOVED to the list.
-     * @param event The event to register.
-     */
-    void registerEvent(Event&& event) { this->tempEvents_.insert(std::move(event)); }
-
-    /**
-     * Actually register events in the permanent list.
-     * @param txHash The hash of the transaction that emitted the events.
-     * @param txIndex The index of the transaction inside the block that emitted the events.
+     * Register the event.
      */
-    void commitEvents(const Hash& txHash, const uint64_t txIndex) {
-      uint64_t logIndex = 0;
-      auto it = tempEvents_.begin(); // Use iterators to loop through the MultiIndex container
-      while (it != tempEvents_.end()) {
-        // Since we can't modify the element directly because iterators are const...
-        Event e = *it;                // ...we make a copy...
-        e.setStateData(logIndex, txHash, txIndex,
-          ContractGlobals::getBlockHash(), ContractGlobals::getBlockHeight()
-        );                            // ...modify it...
-        events_.insert(std::move(e)); // ...move it to the permanent container...
-        it = tempEvents_.erase(it);   // ...erase the original from the temp...
-        logIndex++;                   // ...and move on to the next
-      }
-    }
+    void registerEvent(Event&& event) { this->events_.insert(std::move(event)); }
```

Comment:
Why not receive by value?

```suggestion
    void registerEvent(Event event) { this->events_.insert(std::move(event)); }
```